torch/csrc/inductor/aoti_eager/kernel_holder.cpp

```diff
@@ -81,18 +83,47 @@ std::vector<at::Tensor> unpack_tensors(
   return inputs;
 }
 
+// Find the first positional argument that isn't defaulted
+bool is_default_value(
+    const c10::Argument& argument,
+    const c10::IValue& ivalue) {
+  if (!argument.default_value().has_value()) {
+    return false;
+  }
+  const auto& default_ivalue = *argument.default_value();
+  if (default_ivalue != ivalue) {
+    return false;
+  }
+
+  return true;
+}
+
 std::vector<ParameterMetadata> unpack_input_parameters(
     const std::vector<c10::Argument>& arguments,
     const torch::jit::Stack& stack) {
   std::vector<ParameterMetadata> inputs_metadata;
+  // Represent the order of argument and skip default parameter
+  int64_t arg_order = 0;
   for (size_t idx = 0; idx < stack.size(); idx++) {
+    // By default, the parameter will not be cached if its value is the default
+    // value.
+    //   - produce_aoti_kernel_lib utilizes parseIValuesToPyArgsKwargs to get
+    //   args and kwargs.
+    //   - parseIValuesToPyArgsKwargs skips the parameter if its value is the
+    //   default value.
+    if (is_default_value(arguments[idx], stack[idx])) {
+      continue;
+    }
+
     if (stack[idx].isScalar()) {
-      // scalar
-      inputs_metadata.push_back(ParameterMetadata(stack[idx].toScalar(), idx));
+      // Beyond c10::Scalar, the floating value and interger value are also
+      // represented as Scalar.
+      inputs_metadata.push_back(
+          ParameterMetadata(stack[idx].toScalar(), arg_order));
```

Comment:
I guess it is safer to put `arg_order++` here instead of in the end of the loop, making sure it is also incremented when new arg meta is added.